plugin/core/panels.py

```diff
@@ -183,11 +183,28 @@ def run(self, edit: sublime.Edit, window_id: int) -> None:
                 self.view.insert(edit, self.view.size(), ''.join(new_lines))
                 last_region_end = 0  # Starting from point 0 in the panel ...
                 total_lines, _ = self.view.rowcol(self.view.size())
-                for _ in range(0, max(0, total_lines - SERVER_PANEL_MAX_LINES)):
+                for _ in range(0, max(0, total_lines - LOG_PANEL_MAX_LINES)):
                     # ... collect all regions that span an entire line ...
                     region = self.view.full_line(last_region_end)
                     last_region_end = region.b
                 erase_region = sublime.Region(0, last_region_end)
                 if not erase_region.empty():
                     self.view.erase(edit, erase_region)
         clear_undo_stack(self.view)
+
+
+class LspClearLogPanelCommand(sublime_plugin.TextCommand):
+    def run(self, edit: sublime.Edit) -> None:
+        window = self.view.window()
+        if not window:
+            return
+        panel = ensure_log_panel(window)
+        if panel:
+            panel.run_command("lsp_clear_panel")
+
+    def is_visible(self) -> bool:
+        window = self.view.window()
+        if not window:
+            return False
+        panel = ensure_log_panel(window)
+        return bool(panel and panel.id() == self.view.id())
```

Comment:
```suggestion
        return self.view == window.find_output_panel(PanelName.Log)
```

Would be simpler and doesn't create the output panel from `is_visible`.

Reply:
`ensure_log_panel` will first try to `find_output_panel` before it creates a panel.

```
def ensure_panel(window: sublime.Window, name: str, result_file_regex: str, result_line_regex: str,
                 syntax: str) -> Optional[sublime.View]:
    return window.find_output_panel(name) or create_panel(window, name, result_file_regex, result_line_regex, syntax)
```

Reply:
The point is that it will create the panel if it's not open and created yet.
But it's only theoretical issue though as now the panel is created early and always anyway.